docs/.vuepress/theme/components/SearchBox.vue

```diff
@@ -40,17 +59,61 @@
 import { resolveHeaderTitle } from '../utils/util'
 
 export default {
+  props: {
+    isSearchVisible: {
+      type: Boolean,
+      default: false
+    },
+    isSearchFocused: {
+      type: Boolean,
+      default: false
+    },
+    method: { type: Function }
+  },
+
   data() {
     return {
       query: '',
-      focused: false,
       focusIndex: 0
     }
   },
 
+  watch: {
+    isSearchFocused: function() {
+      // watch it
+      this.isSearchFocused &&
+        document.getElementById('main-search-field').focus()
```

Comment:
Could we solve this with CSS instead? E.g. adding a conditional class :class="{ focused: isSearchFocused }"

Reply:
This isn't a css thing, this targeting the input field:
1. User taps "search" on mobile
2. Search opens, with the input focused and as a result their keyboard opens.

I've had to switch to a `display: none` toggle for the search, which introduces issues with this anyway as it prevents the field from being focused. As a result I've stripped all of this out.

On a related note, something like `:class="{ focused: isSearchFocused }"` is now being used. The css `focus-within` selector isn't incredibly well supported so JS is handling this again (without significant changes):
```
searchClasses() {
      return {
        'search-box': true,
        'search-hidden': !this.isSearchVisible,
        'focus-within': this.focused
      }
    },
```